auction-server/src/auction.rs

```diff
@@ -930,15 +940,97 @@ fn extract_account_svm(
+        let account = accounts.get(account_position).ok_or_else(|| {
+            tracing::error!(
+                "Account not found in transaction accounts: {:?} - {}",
+                accounts,
+                account_position,
+            );
+            RestError::BadParameters("Account not found in transaction accounts".to_string())
+        })?;
+
+        return Ok(*account);
+    }
+
+    let account_position_lookups_writable = account_position - accounts.len();
+    match tx_lookup_tables {
+        Some(tx_lookup_tables) => {
+            let writable_accounts: Vec<(Pubkey, u8)> = tx_lookup_tables
+                .iter()
+                .flat_map(|x| {
+                    x.writable_indexes
+                        .clone()
+                        .into_iter()
+                        .map(|y| (x.account_key, y))
+                })
+                .collect();
+            let readable_accounts: Vec<(Pubkey, u8)> = tx_lookup_tables
+                .iter()
+                .flat_map(|x| {
+                    x.readonly_indexes
+                        .clone()
+                        .into_iter()
+                        .map(|y| (x.account_key, y))
+                })
+                .collect();
+
+            if account_position_lookups_writable < writable_accounts.len() {
+                return query_lookup_table(
+                    writable_accounts,
+                    account_position_lookups_writable,
+                    client,
+                )
+                .await;
+            }
+
+            let account_position_lookups_readable =
+                account_position_lookups_writable - writable_accounts.len();
+            query_lookup_table(readable_accounts, account_position_lookups_readable, client).await
+        }
+        None => {
+            tracing::error!("Lookup tables not found where account_position {} exceeds number of static accounts {}", account_position, accounts.len());
+            Err(RestError::BadParameters(
+                "Lookup tables not found in submit_bid instruction".to_string(),
+            ))
+        }
+    }
+}
+
+async fn query_lookup_table(
+    lookup_accounts: Vec<(Pubkey, u8)>,
+    account_position: usize,
+    client: &RpcClient,
+) -> Result<Pubkey, RestError> {
+    let (table_to_query, index_to_query) =
+        lookup_accounts.get(account_position).ok_or_else(|| {
+            tracing::error!(
+                "Lookup table not found in lookup accounts: {:?} - {}",
+                lookup_accounts,
+                account_position,
+            );
+            RestError::BadParameters("".to_string())
+        })?;
 
+    let table_data = client.get_account_data(table_to_query).await.map_err(|e| {
+        tracing::error!("Error while getting lookup table data: {:?}", e);
+        RestError::BadParameters("Error while getting lookup table data".to_string())
+    })?;
+    let table = AddressLookupTable::deserialize(&table_data).map_err(|e| {
```

Comment:
i tried doing this all from the start originally. there were some challenges bc the most natural place to store them was the `ChainStoreSvm`, but because this is referenced via `Arc`, you can't modify anything within that object. worth a little bit more thinking, possibly creating a totally separate lookup cache object, and i think it would be better to have a separate PR to handle all those changes